### PR TITLE
Update rust release checklist to ensure updated dependencies. 

### PR DESCRIPTION
**Pull request by prestist**
Since dependabot does not seem to update dependencies of dependencies and we have CI which checks for any major breaks. Add a `cargo update` step for rust release checklist to reduce stale package references.

### CODE REVIEW

**Review by bgilbert — COMMENTED** [review]
In the commit message, the prefix should be `rust:`.

File: rust/release-checklist.md [hunk 1]
```diff
@@ -38,19 +38,32 @@ Push access to the upstream repository is required in order to publish the new t
 
 :warning:: if `origin` is not the name of the locally configured remote that points to the upstream git repository (i.e. `git@github.com:coreos/{{ git_repo }}.git`), be sure to assign the correct remote name to the `UPSTREAM_REMOTE` variable.
 
+- declare environment variables
+  - [ ] `RELEASE_VER=x.y.z`
+  - [ ] `UPSTREAM_REMOTE=origin`
+
 {% if do_release_notes_doc %}
+- update all dependencies
```
[the remark this answers is not in the document]
Comment (prestist) [reply rust/release-checklist.md:45]: 100%

File: rust/release-checklist.md [hunk 2]
```diff
@@ -38,19 +38,32 @@ Push access to the upstream repository is required in order to publish the new t
...
+  - [ ] `git checkout -b pre-release-${RELEASE_VER}`
+  - [ ] `cargo update`
+  - [ ] `git commit -m "update dependencies for release ${RELEASE_VER}"`
```
[the remark this answers is not in the document]
Comment (prestist) [reply rust/release-checklist.md]: SGTM

File: rust/release-checklist.md [hunk 3]
```diff
@@ -38,19 +38,32 @@ Push access to the upstream repository is required in order to publish the new t
...
+  
 - write release notes
-  - [ ] write release notes in `docs/release-notes.md`; get them reviewed and merged
```
[the remark this answers is not in the document]
Comment (prestist) [reply rust/release-checklist.md:43]: AH, yes. thank you

File: rust/release-checklist.md [hunk 4]
```diff
@@ -38,19 +38,30 @@ Push access to the upstream repository is required in order to publish the new t
...
 {% if do_release_notes_doc %}
+
 - write release notes
-  - [ ] write release notes in `docs/release-notes.md`; get them reviewed and merged
+  - [ ] write release notes in `docs/release-notes.md`
+  - [ ] `git commit -m "docs/release-notes: update for release ${RELEASE_VER}"`
+  - [ ] create a pull request to review and merge these changes.
   - [ ] if doing a branched release, also include a PR to merge the `docs/release-notes.md` changes into main
+{% else %}
+  - [ ]  create a pull request to review and merge these changes.
```
Comment (bgilbert) [inline rust/release-checklist.md:57]:
```suggestion
  - [ ] PR the changes, get them reviewed, approved and merged
```

File: rust/release-checklist.md [hunk 5]
```diff
@@ -38,19 +38,30 @@ Push access to the upstream repository is required in order to publish the new t
...
 :warning:: if `origin` is not the name of the locally configured remote that points to the upstream git repository (i.e. `git@github.com:coreos/{{ git_repo }}.git`), be sure to assign the correct remote name to the `UPSTREAM_REMOTE` variable.
 
+- declare environment variables
+  - [ ] `RELEASE_VER=x.y.z`
+  - [ ] `UPSTREAM_REMOTE=origin`
+
+- update all dependencies
+  - [ ] `git checkout -b pre-release-${RELEASE_VER}`
+  - [ ] `cargo update`
+  - [ ] `git commit -m "cargo: update dependencies"`
```
Comment (bgilbert) [inline rust/release-checklist.md:48]: We need to either use `--all` or specify the files to add.  We haven't cleaned the working directory yet, so the latter is safer:

```suggestion
  - [ ] `git add Cargo.lock && git commit -m "cargo: update dependencies"`
```

File: rust/release-checklist.md [hunk 6]
```diff
@@ -38,19 +38,30 @@ Push access to the upstream repository is required in order to publish the new t
...
 {% if do_release_notes_doc %}
+
 - write release notes
-  - [ ] write release notes in `docs/release-notes.md`; get them reviewed and merged
+  - [ ] write release notes in `docs/release-notes.md`
+  - [ ] `git commit -m "docs/release-notes: update for release ${RELEASE_VER}"`
```
Comment (bgilbert) [inline rust/release-checklist.md:53]:
```suggestion
  - [ ] `git add docs/release-notes.md && git commit -m "docs/release-notes: update for release ${RELEASE_VER}"`
```

File: rust/release-checklist.md (hunk 5 above)
Comment (prestist) [reply rust/release-checklist.md]: Thank you, I had not considered that being a possible area for unwanted changes.

File: rust/release-checklist.md [hunk 7]
```diff
@@ -38,19 +38,30 @@ Push access to the upstream repository is required in order to publish the new t
...
 :warning:: if `origin` is not the name of the locally configured remote that points to the upstream git repository (i.e. `git@github.com:coreos/{{ git_repo }}.git`), be sure to assign the correct remote name to the `UPSTREAM_REMOTE` variable.
 
+- declare environment variables
+  - [ ] `RELEASE_VER=x.y.z`
+  - [ ] `UPSTREAM_REMOTE=origin`
+
+- update all dependencies
+  - [ ] `git checkout -b pre-release-${RELEASE_VER}`
+  - [ ] `cargo update`
+  - [ ] `git add Cargo.lock && git commit -m "cargo: update dependencies"`
```
Comment (bgilbert) [inline rust/release-checklist.md:48]: I forgot that `Cargo.lock` only gets committed to the repo for applications and not for libraries.  So... we'll need to conditionalize this entire section.

**Comment by prestist** [comment]
Will be rebasing shortly

File: rust/release-checklist.md (hunk 7 above)
Comment (prestist) [reply rust/release-checklist.md:49]: Ah great point.

**Review by bgilbert — APPROVED** [review]
Looks good, thanks!

**Pull request merged**